### PR TITLE
enh(openid): Add custom redirect_url - 22.10 (#1276)

### PR DESCRIPTION
**Pull request by dmyios**
## Description

Backport of https://github.com/centreon/centreon/pull/1200 to 22.10

**Fixes** # MON-18051

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x
- [ ] 23.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by Thebarda — APPROVED** [review]

File: centreon/www/install/php/Update-22.10.8.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    $customConfiguration = json_decode($customConfigurationJson, true);
+    if (! array_key_exists('redirect_url', $customConfiguration)) {
+        $customConfiguration['redirect_url'] = null;
+        $updatedCustomConfigurationEncoded = json_encode($customConfiguration);
+
+        $pearDB->query(
+            <<<SQL
+            UPDATE provider_configuration
+                SET custom_configuration = $updatedCustomConfigurationEncoded
```
Comment (dmyios) [inline centreon/www/install/php/Update-22.10.8.php:48]:
```suggestion
                SET custom_configuration = '$updatedCustomConfigurationEncoded'
```

File: centreon/www/install/php/Update-22.10.8.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            SQL
+        );
+    }
+};
+
+try {
+    $errorMessage = 'Unable to update OpenId custom configuration';
```
Comment (dmyios) [inline centreon/www/install/php/Update-22.10.8.php:55]:
```suggestion
    $errorMessage = 'Unable to update provider_configuration table to add redirect_url';
```

File: centreon/www/install/php/Update-22.10.8.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        $customConfiguration['redirect_url'] = null;
+        $updatedCustomConfigurationEncoded = json_encode($customConfiguration);
+
+        $pearDB->query(
+            <<<SQL
+            UPDATE provider_configuration
+                SET custom_configuration = '$updatedCustomConfigurationEncoded'
+            SQL
+        );
+    }
```
Comment (jeremyjaouen) [inline centreon/www/install/php/Update-22.10.8.php:51]: please use a prepare statement

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: centreon/www/install/php/Update-22.10.8.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        $customConfiguration['redirect_url'] = null;
+        $updatedCustomConfigurationEncoded = json_encode($customConfiguration);
+
+        $statement = $pearDB->prepare(
+            <<<SQL
+            UPDATE provider_configuration
+                SET custom_configuration = :encodedConfiguration
+            SQL
+        );
+        $statement->bindValue('encodedConfiguration', $updatedCustomConfigurationEncoded, \PDO::PARAM_STR);
```
Comment (dmyios) [inline centreon/www/install/php/Update-22.10.8.php:51]:
```suggestion
        $statement->bindValue(':encodedConfiguration', $updatedCustomConfigurationEncoded, \PDO::PARAM_STR);
```

File: centreon/www/install/php/Update-22.10.8.php (hunk 3 above)
Comment (dmyios) [reply centreon/www/install/php/Update-22.10.8.php]: Done

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**